pkg/codegen/hcl2/model/type_object.go

```diff
@@ -72,9 +72,27 @@ func (t *ObjectType) Traverse(traverser hcl.Traverser) (Traversable, hcl.Diagnos
 	keyString, err := convert.Convert(key, cty.String)
 	contract.Assert(err == nil)
 
+	propertiesLower := make(map[string]string)
+	for p := range t.Properties {
+		propertiesLower[strings.ToLower(p)] = p
+	}
+
 	propertyName := keyString.AsString()
 	propertyType, hasProperty := t.Properties[propertyName]
 	if !hasProperty {
+		propertyNameLower := strings.ToLower(propertyName)
+		if propertyNameOrig, ok := propertiesLower[propertyNameLower]; ok {
+			propertyType = t.Properties[propertyNameOrig]
+			rng := traverser.SourceRange()
+			return propertyType, hcl.Diagnostics{
+				{
+					Severity: hcl.DiagWarning,
+					Subject:  &rng,
+					Summary:  "Found matching case-insensitive property",
+					Detail:   fmt.Sprintf("Matched %s with %s", propertyName, propertyNameOrig),
+				},
+			}
+		}
```

Comment:
Can add or update one of our testdata .pp files to check the behavior?

Reply:
I think `type_test.go` is actually the place